### PR TITLE
[studio-ui] Update to GraphiQL to the latest version #3341

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/3341

### CODE REVIEW

File: static-assets/components/cstudio-admin/mods/graphiql.js [hunk 1]
```diff
@@ -15,9 +15,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-CStudioAuthoring.Utils.addJavascript("/static-assets/libs/graphiql/graphiql.js");
-CStudioAuthoring.Utils.addCss("/static-assets/libs/graphiql/graphiql.css");		// library css
-CStudioAuthoring.Utils.addCss("/static-assets/components/cstudio-admin/mods/graphiql.css");		// mod css
+CStudioAuthoring.Utils.addCss("/static-assets/styles/graphiql.css");
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/graphiql.js:18]: Merge these two css files into one.

File: static-assets/components/cstudio-admin/mods/graphiql.js [hunk 2]
```diff
@@ -32,8 +31,8 @@ var wfStates = [];
 YAHOO.extend(CStudioAdminConsole.Tool.GraphiQL, CStudioAdminConsole.Tool, {
 	renderWorkarea: function() {
 		var workareaEl = document.getElementById("cstudio-admin-console-workarea");
-		
-		workareaEl.innerHTML = 
+
+		workareaEl.innerHTML =
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/graphiql.js]: `$('#cstudio-admin-console-workarea').html('<div id="graphContainer"/>');`

Notice the single quotes and camelcase id

File: static-assets/components/cstudio-admin/mods/graphiql.js [hunk 3]
```diff
@@ -45,11 +44,16 @@ YAHOO.extend(CStudioAdminConsole.Tool.GraphiQL, CStudioAdminConsole.Tool, {
...
 
 		CStudioAuthoring.Service.lookupConfigurtion(
-			CStudioAuthoringContext.site, 
+			CStudioAuthoringContext.site,
 			"/environment/environment-config.xml", {
 				success: function(config) {
-					graphServerUrl = config['graphql-server-url'];
-					GraphiQL(document.getElementById("graph-container"), graphServerUrl + '/api/1/site/graphql', site);
+          graphServerUrl = config['graphql-server-url'];
+          CrafterCMSNext.render(document.getElementById("graph-container"), 'GraphiQL',
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/graphiql.js]: Single quotes.
Break after `render(` and each param in it's own line.

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-admin/mods/graphiql.js (hunk 1 above)
Comment (rart) [reply static-assets/components/cstudio-admin/mods/graphiql.js:18]: Also, delete `/static-assets/libs/graphiql`

File: static-assets/components/cstudio-admin/mods/graphiql.js (hunk 2 above)
Comment (rart) [reply static-assets/components/cstudio-admin/mods/graphiql.js]: Please do as shown in the comment

File: static-assets/components/cstudio-admin/mods/graphiql.js [hunk 4]
```diff
@@ -45,21 +43,28 @@ YAHOO.extend(CStudioAdminConsole.Tool.GraphiQL, CStudioAdminConsole.Tool, {
...
 
 		CStudioAuthoring.Service.lookupConfigurtion(
-			CStudioAuthoringContext.site, 
-			"/environment/environment-config.xml", {
+			CStudioAuthoringContext.site,
+			'/environment/environment-config.xml', {
 				success: function(config) {
-					graphServerUrl = config['graphql-server-url'];
-					GraphiQL(document.getElementById("graph-container"), graphServerUrl + '/api/1/site/graphql', site);
+          graphServerUrl = config['graphql-server-url'];
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/graphiql.js]: Looks like this is creating a global variable. Don't see a `const`/`let` statement where graphServerUrl is defined.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/GraphiQL.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+function Graphi(props: GraphiQLProps) {
+  const [query, setQuery] = useState(window.localStorage.getItem(`${props.storageKey}graphiql:query`));
+  const [schema, setSchema] = useState<GraphQLSchema>(null);
+  const [explorerIsOpen, setExplorerIsOpen] = useState<boolean>(false);
+  const _graphiql: GraphiQL = useRef();
+
+  // Updates localStorage and query variable used by graphiQL and graphiQL explorer
+  function onEditQuery(newQuery:any) {
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: Is `newQuery` not of type string?

File: ui/app/src/components/GraphiQL.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    setQuery(newQuery);
+    window.localStorage.setItem(`${props.storageKey}graphiql:query`, newQuery);
+  }
+
+  function graphQLFetcher(graphQLParams: any) {
+    var url:string = props.url ? props.url : window.location.origin + '/api/1/site/graphql',
+        method:string = props.method;
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: Do `(props.method || 'post').toLowerCase()`

File: ui/app/src/components/GraphiQL.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+    if('get' === method){
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx:66]: Refactor: extract the `fetch` invocation out of the if else it's mostly duplicate. For the post body, add it conditionally like so:
```js
fetch(url, {
  method,
  headers: {...},
  ...method === 'post' ? { body: JSON... } : {}
}).then(
```

File: ui/app/src/components/GraphiQL.tsx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+  const _graphiql: GraphiQL = useRef();
+
+  // Updates localStorage and query variable used by graphiQL and graphiQL explorer
+  function onEditQuery(newQuery:any) {
+    setQuery(newQuery);
+    window.localStorage.setItem(`${props.storageKey}graphiql:query`, newQuery);
+  }
+
+  function graphQLFetcher(graphQLParams: any) {
+    var url:string = props.url ? props.url : window.location.origin + '/api/1/site/graphql',
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: Space between `:` and types (e.g. `url:string` => `url: string`). Make sure the document is all properly formatted

File: ui/app/src/components/GraphiQL.tsx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+import GraphiQLExplorer from 'graphiql-explorer';
+import { buildClientSchema, getIntrospectionQuery, GraphQLSchema } from 'graphql';
+
+export interface GraphiQLProps {
+  url?: string,
+  storageKey?: string,
+  method?: string
+}
+
+const _storages: any = {};
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: Remove all `_` from function and variable names.

File: ui/app/src/components/GraphiQL.tsx [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+      }).then(result => {
+        setSchema(buildClientSchema(result.data));
+      });
+    },
+    // eslint-disable-next-line
+    []
+  );
+
+  return (
+    <div className='graphiql-container'>
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: Double quotes for HTML (including JSX)

File: static-assets/components/cstudio-admin/mods/graphiql.js [hunk 11]
```diff
@@ -31,35 +29,38 @@ var wfStates = [];
...
+			'/environment/environment-config.xml', {
 				success: function(config) {
-					graphServerUrl = config['graphql-server-url'];
-					GraphiQL(document.getElementById("graph-container"), graphServerUrl + '/api/1/site/graphql', site);
+          const graphServerUrl = config['graphql-server-url'];
+          CrafterCMSNext.render(
+            document.getElementById('graphContainer'),
+            'GraphiQL',
+            {
+              url: graphServerUrl + '/api/1/site/graphql',
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/graphiql.js]: `${graphServerUrl}/api/1/site/graphql`

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/GraphiQL.tsx [hunk 12]
```diff
@@ -72,34 +72,22 @@ function Graphi(props: GraphiQLProps) {
...
-        } catch (error) {
-          return responseBody;
-        }
-      });
     }
+
+    return fetch(url, {
+      method,
+      headers: { 'Content-Type': 'application/json' },
+      body: method === 'post' ? JSON.stringify(graphQLParams) : ''
```
Comment (rart) [inline ui/app/src/components/GraphiQL.tsx]: This way, the body attribute is still present in the request. Do it as previously requested.

**Review by rart — CHANGES_REQUESTED** [review]
`ui/app/src/components/App.tsx` is not meant to be committed. Please roll back.

If you commit the Graphi component, it beats the purpose of making it lazy on the Codebase Bridge

File: ui/app/src/components/GraphiQL.tsx (hunk 12 above)
Comment (jvega190) [reply ui/app/src/components/GraphiQL.tsx]: done!

**Review by rart — CHANGES_REQUESTED** [review]
I still see App.tsx in the PR and don't see the `/static-assets/libs/graphiql/*` files getting deleted.

**Review by rart — APPROVED** [review]

**Pull request merged**